lib/media/media_source_engine.js

```diff
@@ -851,6 +851,10 @@ shaka.media.MediaSourceEngine = class {
         // This won't be known until the first video segment is appended.
         const offset = await this.textSequenceModeOffset_;
         this.textEngine_.setTimestampOffset(offset);
+
+        // Reset the promise in case the timestamp offset changed during
+        // a period/discontinuity transition.
+        this.textSequenceModeOffset_ = new shaka.util.PublicPromise();
```

Comment:
It seems that this would make every single text buffer block on the next video buffer to download.

But what if we:

1. Buffer a lot of video
2. Rewind within the buffered range
3. Enable text or change text languages

In this scenario, text would be blocked until we left the buffered range.

So I don't believe this fix is safe.

If the sequence mode offset for text needs to be reset on discontinuity, there is probably a solution involving discontinuity information.

Reply:
Thanks for the feedback. Yeah the sequence mode text offset needs to be reset on discontinuities. I saw the offset was being calculated on each video segment buffered and thought we might be able to use the current calc in appendBuffer to update the text offset. But the risks called out make sense.

I have an alternate fix that resets the text offset on discontinuities. I'll push that up.